pkg/alertmanager/api.go

```diff
@@ -118,3 +129,48 @@ func (am *MultitenantAlertmanager) DeleteUserConfig(w http.ResponseWriter, r *ht
 
 	w.WriteHeader(http.StatusOK)
 }
+
+func validateUserConfig(cfg alerts.AlertConfigDesc) error {
+	// Validation copied from: https://github.com/prometheus/alertmanager/blob/8e861c646bf67599a1704fc843c6a94d519ce312/cli/check_config.go#L65-L96
+	amCfg, err := config.Load(cfg.RawConfig)
+	if err != nil {
+		return err
+	}
+
+	// Create templates on disk in a temporary directory.
+	// Note: This means the validation will succeed if we can write to tmp but
+	// not to configured data dir, and on the flipside, it'll fail if we can't write
+	// to tmpDir. Ignoring both cases for now as they're ultra rare but will revisit if
+	// we see this in the wild.
+	tmpDir, err := ioutil.TempDir("", "validate-config")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	for _, tmpl := range cfg.Templates {
+		_, err := createTemplatesFile(tmpDir, cfg.User, tmpl.Filename, tmpl.Body)
+		if err != nil {
+			return err
+		}
+	}
+
+	templateFiles := make([]string, len(amCfg.Templates))
+	if len(amCfg.Templates) > 0 {
```

Comment:
You can remove this `if`. The next `for` will just not iterate over `amCfg.Templates` if it's empty.